testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/web/expiration/SessionExpirationTestCase.java

```diff
@@ -434,43 +438,55 @@ public void test(@ArquillianResource(SessionOperationServlet.class) @OperateOnDe
+                    if (!destroyed) {
+                        response = client.execute(new HttpGet(SessionOperationServlet.createGetURI(baseURL, "a", sessionId)));
+                        try {
+                            Assert.assertEquals(HttpServletResponse.SC_OK, response.getStatusLine().getStatusCode());
+                            Assert.assertFalse(response.containsHeader(SessionOperationServlet.RESULT));
+                            Assert.assertTrue(response.containsHeader(SessionOperationServlet.SESSION_ID));
+                            Assert.assertEquals(newSessionId == null, response.containsHeader(SessionOperationServlet.CREATED_SESSIONS));
+                            if (newSessionId == null) {
+                                newSessionId = response.getFirstHeader(SessionOperationServlet.SESSION_ID).getValue();
+                            } else {
+                                Assert.assertEquals(newSessionId, response.getFirstHeader(SessionOperationServlet.SESSION_ID).getValue());
+                            }
+                            destroyed = response.containsHeader(SessionOperationServlet.DESTROYED_SESSIONS);
+                            Assert.assertFalse(response.containsHeader(SessionOperationServlet.ADDED_ATTRIBUTES));
+                            Assert.assertFalse(response.containsHeader(SessionOperationServlet.REPLACED_ATTRIBUTES));
+                            Assert.assertEquals(destroyed, response.containsHeader(SessionOperationServlet.REMOVED_ATTRIBUTES));
+                            Assert.assertFalse(response.containsHeader(SessionOperationServlet.BOUND_ATTRIBUTES));
+                            Assert.assertEquals(destroyed, response.containsHeader(SessionOperationServlet.UNBOUND_ATTRIBUTES));
+                            if (destroyed) {
+                                Assert.assertEquals(sessionId, response.getFirstHeader(SessionOperationServlet.DESTROYED_SESSIONS).getValue());
+                                Assert.assertEquals("a", response.getFirstHeader(SessionOperationServlet.REMOVED_ATTRIBUTES).getValue());
+                                Assert.assertEquals("7", response.getFirstHeader(SessionOperationServlet.UNBOUND_ATTRIBUTES).getValue());
+                                log.infof("Session destroyed within %d attempts/seconds.", attempt);
+                            }
+                        } finally {
+                            HttpClientUtils.closeQuietly(response);
                         }
-                    } finally {
-                        HttpClientUtils.closeQuietly(response);
                     }
+                    // Wait a second between retries
+                    Thread.sleep(TimeUnit.SECONDS.toMillis(1));
                 }
             }
-            Assert.assertTrue(destroyed);
+            Assert.assertTrue("Session has not been destroyed following expiration within " + maxAttempts + " seconds.", destroyed);
```

Comment:
The assertion message needs to be updated.

Reply:
How so?

Reply:
maxAttempts is not a measure of seconds.

Reply:
Well, each attempt takes a second, so it is... changed wording to attempts if its causing confusion. Anyway, any message is better than no message since line numbers change one needs to look into the code to see when the test actually failed.